src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WebSocketHandle.Unix.cs

```diff
@@ -1358,7 +1358,123 @@ private struct MessageHeader
                 public bool Fin;
                 public long PayloadLength;
             }
+
+            /// <summary>
+            /// A custom network stream that stores and reuses a single SocketAsyncEventArgs instance
+            /// for reads and a single SocketAsyncEventArgs instance for writes.  This limits it to
+            /// supporting a single read and a single write at a time, but with much less per-operation
+            /// overhead than with System.Net.Sockets.NetworkStream.
+            /// </summary>
+            private sealed class AsyncEventArgsNetworkStream : NetworkStream
+            {
+                private readonly Socket _socket;
+                private readonly SocketAsyncEventArgs _readArgs;
+                private readonly SocketAsyncEventArgs _writeArgs;
+
+                private AsyncTaskMethodBuilder<int> _readTcs;
+                private AsyncTaskMethodBuilder _writeTcs;
```

Comment:
What's "TCS"?  I'm auto-expanding it as "task completion source", but this is a task factory, sort of the opposite of that.